packages/job-worker/src/blueprints/context/SyncIngestUpdateToPartInstanceContext.ts

```diff
@@ -221,6 +221,14 @@ export class SyncIngestUpdateToPartInstanceContext
 
 		if (!this.partInstance) throw new Error(`PartInstance has been removed`)
 
+		// for autoNext, the new expectedDuration cannot be shorter than the time a part has been on-air for
+		if (trimmedProps.expectedDuration && (trimmedProps.autoNext ?? this.partInstance.part.autoNext)) {
+			const onAir = this.partInstance.timings?.reportedStartedPlayback
+			if (onAir && Date.now() - onAir > trimmedProps.expectedDuration) {
+				trimmedProps.expectedDuration = Date.now() - onAir
```

Comment:
I think it would be good to add a tolerance here. So that it sets the end to now + 500ms or something.  
My concern is that when this runs, there is still often 80ms+ until TSR has resolved this timeline, so it wont be a bad jump, but TSR will still be playing catchup.

Reply:
I agree and had a todo initially, but then I realised it's a bit of a magic number and I'm not sure how to handle that in a good way

Reply:
We already some magic constants in the code, doing things like blocking the next part from changing when within a second of an autonext, so Im not opposed to this being hardcoded.

Alternatively, maybe this should be a setting on the studio? That makes it slightly less hardcoded.
The optimal solution would be something using the latency reported by playout-gateway, but we would still need to factor in the time taken by core, so will be hard to do